module/zfs/zfs_vnops.c

```diff
@@ -1446,10 +1447,20 @@ zfs_create(struct inode *dip, char *name, vattr_t *vap, int excl,
 		}
 		zfs_mknode(dzp, vap, tx, cr, 0, &zp, &acl_ids);
 
+		error = zfs_link_create(dl, zp, tx, ZNEW);
+		if (error != 0) {
+			/*
+			 * Delete the node created above.
+			 */
```

Comment:
[nit]  I'd drop the comment here or update it to be clear why `zp` from `zfs_mknode` needs to be destroyed.

Reply:
Have reworded the comment. If does not serve the purpose, I am fine with deleting it.